src/extensions/content_downloaders/curl_downloader/curl_content_downloader.EXPORTS.cpp

```diff
@@ -28,6 +30,15 @@ ADUC_Result Download(
     unsigned int timeoutInSeconds,
     ADUC_DownloadProgressCallback downloadProgressCallback)
 {
+    // JIT init since Initialize is called by main thread and
```

Comment:
not a fan of this.  I'd prefer to see a extension_Initialize() (and Uninitialize() ) function where this is called and uninitialized (as needed).  Relying on Download means there maybe be cases where logging cant occur in an extension

Reply:
Agree. I think I could add a download_Init() and download_Uninit() that is called on download thread creation and destruction, respectively.

Reply:
Added OnDownloadBegin and OnDownloadEnd.  It was missing a couple of traces earlier. It gets the following now in `/var/log/adu/do-content-downloader.*.log`:

```sh
2023-03-28T02:04:10.8268Z 315409[315415] [I] [do_download] ==== MULTI-LINE LOG BEGIN ====
Downloading File 'apt-manifest.json' from 'http://jewelden2--jewelden2.b.nlu.dl.adu.microsoft.com/eastus/jewelden2--jewelden2/272aeacb92594d51a422ed8d60f3a0c5/apt-manifest.json' to '/var/lib/adu/downloads/5935ab72-9122-4b7a-9405-a5b860b9bd87/apt-manifest.json'2023-03-28T02:04:10.8268Z 315409[315415] [I] [do_download] ==== MULTI-LINE LOG END ====

2023-03-28T02:04:11.5878Z 315409[315415] [I] Validating file hash [do_download]
2023-03-28T02:04:11.5880Z 315409[315415] [I] Download resultCode: 500, extendedCode: 0 [do_download]
```